tests/mr/device/mr_test.hpp

```diff
@@ -49,11 +50,25 @@ namespace rmm::test {
  * @brief Returns if a pointer points to a device memory or managed memory
  * allocation.
  */
-inline bool is_device_memory(void* ptr)
+inline bool is_device_accessible_memory(void* ptr)
 {
   cudaPointerAttributes attributes{};
   if (cudaSuccess != cudaPointerGetAttributes(&attributes, ptr)) { return false; }
-  return (attributes.type == cudaMemoryTypeDevice) or (attributes.type == cudaMemoryTypeManaged);
+  return (attributes.type == cudaMemoryTypeDevice) or (attributes.type == cudaMemoryTypeManaged) or
+         ((attributes.type == cudaMemoryTypeHost) and (attributes.devicePointer != nullptr));
+}
+
+inline bool is_host_memory(void* ptr)
```

Comment:
It feels odd that we're copying these functions in multiple headers rather than defining them in a single place and including them. Is there a reason for that?

Reply:
Refactored. Thanks for catching this.